src/importlinter/contracts/layers.py

```diff
@@ -39,19 +39,29 @@ class LayersContract(Contract):
 
     Configuration options:
 
-        - layers:         An ordered list of layers. Each layer is the name of a module relative
-                          to its parent package. The order is from higher to lower level layers.
-        - containers:     A list of the parent Modules of the layers (optional).
-        - ignore_imports: A set of DirectImports. These imports will be ignored: if the import
-                          would cause a contract to be broken, adding it to the set will cause
-                          the contract be kept instead. (Optional.)
+        - layers:             An ordered list of layers. Each layer is the name of a module
+                              relative to its parent package. The order is from higher to lower
+                              level layers.
+        - containers:         A list of the parent Modules of the layers (optional).
+        - ignore_imports:     A set of DirectImports. These imports will be ignored: if the import
+                              would cause a contract to be broken, adding it to the set will cause
+                              the contract be kept instead. (Optional.)
+        - exhaustive:         If true, check that the contract declares every possible layer in
+                              its list of layers to check. This option cannot be used without the
+                              containers option. (Optional.)
+        - exhaustive_ignores: A set of potential layers to ignore in exhaustiveness checks. The
+                              presence of this option implies exhaustive. (Optional.)
     """
 
     type_name = "layers"
 
     layers = fields.ListField(subfield=LayerField())
     containers = fields.ListField(subfield=fields.StringField(), required=False)
     ignore_imports = fields.SetField(subfield=fields.DirectImportField(), required=False)
+    exhaustive = fields.BooleanField(required=False, default=False)
+    exhaustive_ignores = fields.SetField(
+        subfield=fields.StringField(), required=False, default=set
```

Comment:
NIP: It's generally a good approach to use immutable containers as parameter defaults, so in this case e.g:

```suggestion
        subfield=fields.StringField(), required=False, default=frozenset()
```

Reply:
I don't have a strong opinion either way, so I would be happy to make the change, but what about types that don't have immutable versions?

I'm thinking mostly of `list` here - you can use `tuple` as an immutable sequence, but it isn't strictly a subtype of `list` like `frozenset` is of `set`.